### DISCUSSION TITLE
[OOP] Task await until IRQ is processed, how to pass data IRQ to task

### DISCUSSION THREAD

**Discussion by hitech95** [arkhipenko/TaskScheduler#162] [Q&A] [answered]
Hi,
I have a device that receive an interrupt, I then read a specific register with the interrupt reason.
I have to read it as soon as possible otherwise it can change.

After reading the register I have to schedule a Task to do processing with this read data.

Right now I'm doing:
```
// Called by the library after reading the IRQ register
bool onStatusUpdate(uint8_t error)
{
    // FIXME - pass error to taskIRQ
    return taskIRQ->restart();
}
```

What would be the best way to achieve so?
I've tried with a StatusRequest but no luck so far.

**Comment by arkhipenko** [comment, marked as answer]
Hi @hitech95 
From the scheduling perspective (meaning, scheduling the task to run ASAP) you are doing everything right - scheduling task to restart during interrupt. Depending where in the execution chain taskIRQ is, it might not be the very next task to be activated. You might want to assign those to a scheduler of a higher priority, then they will be evaluated for invocation more frequently. 
Example here: https://github.com/arkhipenko/TaskScheduler/blob/master/examples/Scheduler_example11_Priority/Scheduler_example11_Priority.ino

From the passing a value to the task perspective, you need to use LTS (local thread storage)
Example here: https://github.com/arkhipenko/TaskScheduler/blob/master/examples/Scheduler_example08_LTS/Scheduler_example08_LTS.ino

https://github.com/arkhipenko/TaskScheduler/wiki/API-Task#void-setltspointervoid-aptr


Every task has a 4 byte area to either store a int32 value or a pointer to a data structure where the data is. 
If all you need is a pass a byte, you can just use the LTS area to pass value itself, not a pointer. 

```
uint8_t reason = interrupt.getReason(); //or whatever
taskIRQ->setLtsPointer((void*) reason); // something along those lines - you are passing the `reason` value as if it was a pointer address, but you are not going to use it that way. may need to go via an actual pointer
```
